lua_examples/luaOTA/_doTick.lua

```diff
@@ -0,0 +1,76 @@
+tmr.stop(0)--SAFETRIM 
+-- function _doTick(self)
+
+  -- Upvals
+  local self                = ...
+  local wifi,net            = wifi,net 
+  local sta                 = wifi.sta
+  local config,log,startApp = self.config,self.log,self.startApp
+  local tick_count          = 0
+  
+  local function socket_close(socket) --upval: self, startApp
+    if rawget(self,"socket") then
+      self.socket=nil                        -- remove circular reference in upval 
+      pcall(socket.close,socket)
+      return startApp("Unexpected socket close")
+    end
+  end
+
+  local function receiveFirstRec(socket, rec)  -- upval: self, crypto, startApp, tmr
+    local cmdlen   = (rec:find('\n',1, true) or 0) - 1
+    local cmd,hash = rec:sub(1,cmdlen-6), rec:sub(cmdlen-5,cmdlen)
+    if cmd:find('"r":"OK!"',1,true) or cmdlen < 16 or
+      hash ~= crypto.toHex(crypto.hash("MD5",self.secret .. cmd):sub(-3)) then
```

Comment:
@nwf Nathan, yes MD5 is weak hash and I would no longer use it for password mangling on an application such as mediaWiki or a forum package such as phpBB3 ot InvsionPower (and I administer all of these, BTW), but it is only weak in the sense that it can be brute-force attacked in multistream processors such as a CUDA architecture; hack in and get a password file of say 10K accounts then the attacker would probably crack perhaps 30-50% of typical user passwords in an 24hr compute.  If the IoT device and the server used say "wTRfe$523'dfaw;WEqwe2" then hell would freeze over before the attacker cracked it. Yuo don't have scale advantages of the "attack N thousand at once).  Incidentally  I originally used SHA1, but MD5 is in most Lua PC and \*nix repositories,  SHA1 isn't and I wanted to keep this simple.